operator/src/main/java/oracle/kubernetes/operator/helpers/PodHelper.java

```diff
@@ -452,16 +455,34 @@ String getServerName() {
     @Override
     // let the pod rolling step update the pod
     Step replaceCurrentPod(V1Pod pod, Step next) {
-      return createDomainRollStartEventIfNeeded(pod, deferProcessing(createCyclePodStep(pod, next)));
+      return createDomainRollStartEventIfNeeded(pod, deferProcessing(pod, createCyclePodStep(pod, next)));
     }
 
-    private Step deferProcessing(Step deferredStep) {
+    private Step deferProcessing(V1Pod pod, Step deferredStep) {
       synchronized (packet) {
         Optional.ofNullable(getServersToRoll()).ifPresent(r -> r.put(getServerName(), createRollRequest(deferredStep)));
       }
-      return null;
+      return createPodNeedsToRollStep(pod);
     }
 
+    // Patch the pod to indicate a pending roll.
+    private Step createPodNeedsToRollStep(V1Pod pod) {
+      return isPodAlreadyLabeledToRoll(pod) ? null : createPatchStep();
+    }
+
+    private boolean isPodAlreadyLabeledToRoll(V1Pod pod) {
+      return pod.getMetadata().getLabels().containsKey(LabelConstants.TO_BE_ROLLED_LABEL);
+    }
+
+    private Step createPatchStep() {
+      JsonPatchBuilder patchBuilder = Json.createPatchBuilder();
+      patchBuilder.add("/metadata/labels/" + LabelConstants.TO_BE_ROLLED_LABEL, "true");
+      return new CallBuilder()
+          .patchPodAsync(getPodName(), getNamespace(), getDomainUid(),
```

Comment:
I am wondering if we should use a sync call here to avoid a possible race condition.

Reply:
What kind of race condition are you thinking of?

Reply:
I see; if we are also going to roll that particular pod, it could be bad to add this label afterwards. Looking into making it a synchronous call.

Reply:
@doxiao Patching the pod synchronously works, and should avoid the problem. Please take a look.